hoodie-client/src/main/java/com/uber/hoodie/CompactionAdminClient.java

```diff
@@ -149,18 +149,19 @@ public CompactionAdminClient(JavaSparkContext jsc, String basePath) {
    * @param skipValidation Skip validation
    * @param dryRun         Dry Run Mode
    */
-  public List<RenameOpResult> unscheduleCompactionFileId(String fileId,
+  public List<RenameOpResult> unscheduleCompactionFileId(String partitionPath, String fileId,
```

Comment:
@n3nash : I  can add a new class called HoodieFileId which encapsulates partition-path and fileId. I am skeptical about the other option to concatenate partition + fileId and treat it as one string. Looking at the number of places we need both partition + fileId,, I feel we might shoot ourselves in the foot if we do not distinguish them ( fileId vs partitionPath + fileId) as different types.  Let me know your thoughts.

Reply:
Okay, I looked through the code and I think what you are suggesting makes sense. Can we have the `HoodieFileId` an inner class in FileSlice and then instead of Pair we use `HoodieFileId` as the object to compare on. I was also wondering for all the operations such as `generateCompactionPlan` which takes `fileIdsWithPartitionPathInPendingCompactions` will change to `hoodieFileIdsInPendingCompactions` correct ?

Reply:
@n3nash : Addressed comment. I ended up naming the class as HoodieFileGroupId as it essentially identifies a file-group uniquely. I have also added factories in HoodieFileGroup/FileSlice/CompactionOperation classes as these are the common places from where the unique-id is needed. Please take a look